### PR TITLE
feat(server): Pandas filter step (only simple equality for now)

### PR DESCRIPTION
**Pull request by davinov**
Start of the implementation for the filter step.
The aim of this PR is not to implement all cases of conditions, but to structure the repository for easy step development.
Hence:
- new package to expose steps
- new package for their tests
- pipeline class to execute as many steps as we want

### CODE REVIEW

File: server/weaverbird/steps/filter.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+ColumnName = Union[str, int, float]
+DataValue = Union[bool, float, int, str, list, None]
+
+
+class SimpleCondition(BaseModel):
+    column: ColumnName
+    # TODO support all operators
+    operator: Literal['eq']
+    value: DataValue
```
Comment (fspot) [inline server/weaverbird/steps/filter.py:17]: :warning: attention on a spotté un souci dans laputa : pydantic cast 0 en False à cause de l'ordre de l'Union. Changer l'ordre ne suffit pas car alors il cast False en 0. Le fix : https://github.com/ToucanToco/laputa/pull/3718/commits/5b1f9894b7754acb7354761121cb8c50a7e7ff84

Comment (PrettyWood) [reply server/weaverbird/steps/filter.py:17]: Yes c'est un sujet que je veux pousser pour pydantic 1.8. Top le workaround @fspot !

**Review by fspot — CHANGES_REQUESTED** [review]
Juste une remarque sur le SimpleCondition, si on peut ajouter un test sur ce cas btw ça serait bien :+1:

File: server/weaverbird/steps/filter.py (hunk 1 above)
Comment (davinov) [reply server/weaverbird/steps/filter.py:17]: Oh! :hushed: Nicely spotted

**Comment by davinov** [comment]
@fspot Feebdack applied and rebased on master :)

File: server/weaverbird/steps/filter.py (hunk 1 above)
Comment (davinov) [reply server/weaverbird/steps/filter.py:18]: https://github.com/ToucanToco/weaverbird/pull/671/commits/b24e8d9a1963f264bf3f09b56c3da5d94d4e8553

**Review by fspot — APPROVED** [review]

**Pull request merged**